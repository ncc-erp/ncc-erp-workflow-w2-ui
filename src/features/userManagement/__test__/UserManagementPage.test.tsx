import { render } from '@testing-library/react';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { RecoilRoot } from 'recoil';
import UserManagement from '..';
import { BrowserRouter as Router } from 'react-router-dom';

jest.mock('../../../api/apiHooks/index', () => ({
  useAxios: jest.fn(),
}));

jest.mock('../../../api/axiosInstant', () => ({
  VITE_API_BASE_URL: '/api',
}));

jest.mock('hooks/useIsAdmin', () => ({
  useIsAdmin: jest.fn().mockReturnValue(true),
}));

jest.mock('hooks/useMediaQuery', () => ({
  useMediaQuery: jest.fn().mockReturnValue(true),
}));

jest.mock('api/apiHooks/userIdentityHooks', () => ({
  useUserIdentity: jest.fn().mockReturnValue({
    isLoading: false,
    data: {
      totalCount: 1,
      items: [
        {
          tenantId: null,
          userName: 'admin',
          name: 'admin',
          surname: null,
          email: 'admin@abp.io',
          emailConfirmed: false,
          phoneNumber: null,
          phoneNumberConfirmed: false,
          isActive: true,
          lockoutEnabled: true,
          lockoutEnd: null,
          concurrencyStamp: 'd4d233a94eb54d2da495832e1f8e7c74',
          isDeleted: false,
          deleterId: null,
          deletionTime: null,
          lastModificationTime: '2022-07-28T02:38:42.821371Z',
          lastModifierId: null,
          creationTime: '2022-07-18T09:04:44.590654Z',
          creatorId: null,
          id: '3a0522ba-55ac-e721-2020-8c24bc3e1856',
          extraProperties: {},
        },
      ],
    },
  }),
}));

test('Request My Requests Page', () => {
  const queryClient: QueryClient = new QueryClient();
  const { container } = render(
    <QueryClientProvider client={queryClient}>
      <RecoilRoot>
        <Router>
          <UserManagement />
        </Router>
      </RecoilRoot>
    </QueryClientProvider>
  );
  expect(container).toMatchSnapshot();
});
