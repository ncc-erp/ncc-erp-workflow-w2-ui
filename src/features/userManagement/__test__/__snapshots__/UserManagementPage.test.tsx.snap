// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Request My Requests Page 1`] = `
<div>
  <div
    class="css-0"
  >
    <div
      class="chakra-stack css-jrkth6"
    >
      <div
        class="chakra-stack__item css-1stbcqi"
      >
        <h1
          class="chakra-heading css-ftu039"
        >
          User management
        </h1>
      </div>
      <div
        class="chakra-stack__item css-bdvyq0"
      >
        <h2
          class="chakra-heading css-x3ytmx"
        />
        <div>
          <button
            class="chakra-button css-fkrxci"
            title="Log out"
            type="button"
          >
            <svg
              aria-hidden="true"
              fill="none"
              height="20px"
              stroke="currentColor"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              width="20px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5.636 5.636a9 9 0 1012.728 0M12 3v9"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
          </button>
          <button
            class="chakra-button css-fkrxci"
            title="Release note"
            type="button"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              height="20px"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="20px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M5.625 1.5H9a3.75 3.75 0 013.75 3.75v1.875c0 1.036.84 1.875 1.875 1.875H16.5a3.75 3.75 0 013.75 3.75v7.875c0 1.035-.84 1.875-1.875 1.875H5.625a1.875 1.875 0 01-1.875-1.875V3.375c0-1.036.84-1.875 1.875-1.875zm6.905 9.97a.75.75 0 00-1.06 0l-3 3a.75.75 0 101.06 1.06l1.72-1.72V18a.75.75 0 001.5 0v-4.19l1.72 1.72a.75.75 0 101.06-1.06l-3-3z"
                fill-rule="evenodd"
              />
              <path
                d="M14.25 5.25a5.23 5.23 0 00-1.279-3.434 9.768 9.768 0 016.963 6.963A5.23 5.23 0 0016.5 7.5h-1.875a.375.375 0 01-.375-.375V5.25z"
              />
            </svg>
          </button>
          <button
            class="chakra-button css-fkrxci"
            title="User guide"
            type="button"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              height="20px"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 20 20"
              width="20px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M4 4a2 2 0 012-2h4.586A2 2 0 0112 2.586L15.414 6A2 2 0 0116 7.414V16a2 2 0 01-2 2H6a2 2 0 01-2-2V4zm2 6a1 1 0 011-1h6a1 1 0 110 2H7a1 1 0 01-1-1zm1 3a1 1 0 100 2h6a1 1 0 100-2H7z"
                fill-rule="evenodd"
              />
            </svg>
          </button>
          <button
            class="chakra-button css-v0af4h"
            type="button"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 160c-52.9 0-96 43.1-96 96s43.1 96 96 96 96-43.1 96-96-43.1-96-96-96zm246.4 80.5l-94.7-47.3 33.5-100.4c4.5-13.6-8.4-26.5-21.9-21.9l-100.4 33.5-47.4-94.8c-6.4-12.8-24.6-12.8-31 0l-47.3 94.7L92.7 70.8c-13.6-4.5-26.5 8.4-21.9 21.9l33.5 100.4-94.7 47.4c-12.8 6.4-12.8 24.6 0 31l94.7 47.3-33.5 100.5c-4.5 13.6 8.4 26.5 21.9 21.9l100.4-33.5 47.3 94.7c6.4 12.8 24.6 12.8 31 0l47.3-94.7 100.4 33.5c13.6 4.5 26.5-8.4 21.9-21.9l-33.5-100.4 94.7-47.3c13-6.5 13-24.7.2-31.1zm-155.9 106c-49.9 49.9-131.1 49.9-181 0-49.9-49.9-49.9-131.1 0-181 49.9-49.9 131.1-49.9 181 0 49.9 49.9 49.9 131.1 0 181z"
              />
            </svg>
            <p
              class="chakra-text css-14gt3bc"
            >
              Light
            </p>
          </button>
        </div>
      </div>
    </div>
    <div
      class="css-0"
    >
      <div
        class="css-0"
      >
        <div
          class="chakra-stack css-55unqk"
        >
          <div
            class="chakra-stack css-1uwl1lo"
          >
            <div
              class="chakra-input__group css-q5l91q"
              data-group="true"
            >
              <input
                class="chakra-input css-1t4gsb0"
                placeholder="Enter email"
                type="text"
                value=""
              />
              <div
                class="chakra-input__right-element css-yn8497"
              >
                <svg
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                    stroke="none"
                  />
                  <path
                    d="M10 10m-7 0a7 7 0 1 0 14 0a7 7 0 1 0 -14 0"
                  />
                  <path
                    d="M21 21l-6 -6"
                  />
                </svg>
              </div>
            </div>
            <div
              class="css-0"
            >
              <div
                class="chakra-form-control css-0"
                role="group"
              >
                <div
                  class="chakra-select__wrapper css-42b2qy"
                >
                  <select
                    class="chakra-select css-lcm62o"
                    id="field-:r0:"
                  />
                  <div
                    class="chakra-select__icon-wrapper css-162mkon"
                  >
                    <svg
                      aria-hidden="true"
                      class="chakra-icon chakra-select__icon css-13otjrl"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      role="presentation"
                      stroke="currentColor"
                      stroke-width="0"
                      style="color: currentColor; width: 1em; height: 1em;"
                      viewBox="0 0 24 24"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M12 14L8 10H16L12 14Z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <button
            aria-label="Done"
            class="chakra-button css-lp5hqr"
            type="button"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              focusable="false"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 1024 1024"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M909.1 209.3l-56.4 44.1C775.8 155.1 656.2 92 521.9 92 290 92 102.3 279.5 102 511.5 101.7 743.7 289.8 932 521.9 932c181.3 0 335.8-115 394.6-276.1 1.5-4.2-.7-8.9-4.9-10.3l-56.7-19.5a8 8 0 0 0-10.1 4.8c-1.8 5-3.8 10-5.9 14.9-17.3 41-42.1 77.8-73.7 109.4A344.77 344.77 0 0 1 655.9 829c-42.3 17.9-87.4 27-133.8 27-46.5 0-91.5-9.1-133.8-27A341.5 341.5 0 0 1 279 755.2a342.16 342.16 0 0 1-73.7-109.4c-17.9-42.4-27-87.4-27-133.9s9.1-91.5 27-133.9c17.3-41 42.1-77.8 73.7-109.4 31.6-31.6 68.4-56.4 109.3-73.8 42.3-17.9 87.4-27 133.8-27 46.5 0 91.5 9.1 133.8 27a341.5 341.5 0 0 1 109.3 73.8c9.9 9.9 19.2 20.4 27.8 31.4l-60.2 47a8 8 0 0 0 3 14.1l175.6 43c5 1.2 9.9-2.6 9.9-7.7l.8-180.9c-.1-6.6-7.8-10.3-13-6.2z"
              />
            </svg>
          </button>
        </div>
        <div
          class="css-12d0673"
        >
          <table
            class="chakra-table css-zy166a"
          >
            <thead
              class="css-0"
            >
              <tr
                class="css-1y39i0r"
              >
                <th
                  class="css-175w36i"
                  colspan="1"
                  style="width: auto;"
                >
                  <div
                    class="css-1qe39um"
                  >
                    <div
                      class="css-0"
                    >
                      User name
                    </div>
                    <svg
                      class="chakra-icon css-1pr7mfq"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 512 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M128 320l128-128 128 128z"
                      />
                    </svg>
                  </div>
                </th>
                <th
                  class="css-175w36i"
                  colspan="1"
                  style="width: auto;"
                >
                  <div
                    class="css-1qe39um"
                  >
                    Email address
                    <span
                      style="width: 16px;"
                    />
                  </div>
                </th>
                <th
                  class="css-175w36i"
                  colspan="1"
                  style="width: auto;"
                >
                  <div
                    class="css-1qe39um"
                  >
                    Phone number
                    <span
                      style="width: 16px;"
                    />
                  </div>
                </th>
                <th
                  class="css-175w36i"
                  colspan="1"
                  style="width: auto;"
                >
                  <div
                    class="css-1qe39um"
                  >
                    Roles
                    <span
                      style="width: 16px;"
                    />
                  </div>
                </th>
                <th
                  class="css-avwfu8"
                  colspan="1"
                  style="width: auto;"
                >
                  <div
                    class="css-alaj7i"
                  >
                    <div
                      class="css-mgpx6q"
                    >
                      Actions
                    </div>
                  </div>
                </th>
              </tr>
            </thead>
            <tbody
              class="css-0"
            >
              <tr
                class="css-11yb80h"
              >
                <td
                  class="css-xk16pm"
                >
                  <div
                    class="css-0"
                  >
                     
                    admin
                  </div>
                </td>
                <td
                  class="css-xk16pm"
                >
                  admin@abp.io
                </td>
                <td
                  class="css-xk16pm"
                />
                <td
                  class="css-xk16pm"
                >
                  N/A
                </td>
                <td
                  class="css-xk16pm"
                >
                  <div
                    class="css-gmuwbf"
                  >
                    <button
                      aria-controls="menu-list-:r6:"
                      aria-expanded="false"
                      aria-haspopup="menu"
                      aria-label=""
                      class="chakra-button chakra-menu__menu-button css-p7mkev"
                      id="menu-button-:r6:"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="chakra-icon css-bzk4kq"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        stroke="currentColor"
                        stroke-width="0"
                        viewBox="0 0 24 24"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M5.33409 4.54491C6.3494 3.63637 7.55145 2.9322 8.87555 2.49707C9.60856 3.4128 10.7358 3.99928 12 3.99928C13.2642 3.99928 14.3914 3.4128 15.1245 2.49707C16.4486 2.9322 17.6506 3.63637 18.6659 4.54491C18.2405 5.637 18.2966 6.90531 18.9282 7.99928C19.5602 9.09388 20.6314 9.77679 21.7906 9.95392C21.9279 10.6142 22 11.2983 22 11.9993C22 12.7002 21.9279 13.3844 21.7906 14.0446C20.6314 14.2218 19.5602 14.9047 18.9282 15.9993C18.2966 17.0932 18.2405 18.3616 18.6659 19.4536C17.6506 20.3622 16.4486 21.0664 15.1245 21.5015C14.3914 20.5858 13.2642 19.9993 12 19.9993C10.7358 19.9993 9.60856 20.5858 8.87555 21.5015C7.55145 21.0664 6.3494 20.3622 5.33409 19.4536C5.75952 18.3616 5.7034 17.0932 5.0718 15.9993C4.43983 14.9047 3.36862 14.2218 2.20935 14.0446C2.07212 13.3844 2 12.7002 2 11.9993C2 11.2983 2.07212 10.6142 2.20935 9.95392C3.36862 9.77679 4.43983 9.09388 5.0718 7.99928C5.7034 6.90531 5.75952 5.637 5.33409 4.54491ZM13.5 14.5974C14.9349 13.7689 15.4265 11.9342 14.5981 10.4993C13.7696 9.0644 11.9349 8.57277 10.5 9.4012C9.06512 10.2296 8.5735 12.0644 9.40192 13.4993C10.2304 14.9342 12.0651 15.4258 13.5 14.5974Z"
                        />
                      </svg>
                    </button>
                    <div
                      class="css-0"
                      style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
                    >
                      <div
                        aria-orientation="vertical"
                        class="chakra-menu__menu-list css-1oaq8ss"
                        id="menu-list-:r6:"
                        role="menu"
                        style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
                        tabindex="-1"
                      >
                        <button
                          class="chakra-menu__menuitem css-zn25ca"
                          data-index="0"
                          id="menu-list-:r6:-menuitem-:r7:"
                          role="menuitem"
                          tabindex="-1"
                          type="button"
                        >
                          <svg
                            class="chakra-icon css-zvw1k8"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            stroke="currentColor"
                            stroke-width="0"
                            viewBox="0 0 24 24"
                            width="1em"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M9.24264 18.9964H21V20.9964H3V16.7538L12.8995 6.85431L17.1421 11.0969L9.24264 18.9964ZM14.3137 5.44009L16.435 3.31877C16.8256 2.92825 17.4587 2.92825 17.8492 3.31877L20.6777 6.1472C21.0682 6.53772 21.0682 7.17089 20.6777 7.56141L18.5563 9.68273L14.3137 5.44009Z"
                            />
                          </svg>
                          Edit
                        </button>
                      </div>
                    </div>
                  </div>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
        <div
          class="chakra-stack css-11un3ql"
        >
          <div
            class="chakra-stack css-dvyqh9"
          >
            <div
              class="chakra-stack css-1hohgv6"
            >
              <p
                class="chakra-text css-1ok354g"
              >
                Rows per page
              </p>
              <div
                class="chakra-form-control css-0"
                role="group"
              >
                <div
                  class="chakra-select__wrapper css-bo0or5"
                >
                  <select
                    class="chakra-select css-1y1ernp"
                    id="field-:r4:"
                  >
                    <option
                      value="10"
                    >
                      10
                    </option>
                    <option
                      value="25"
                    >
                      25
                    </option>
                    <option
                      value="50"
                    >
                      50
                    </option>
                    <option
                      value="100"
                    >
                      100
                    </option>
                  </select>
                  <div
                    class="chakra-select__icon-wrapper css-10n8c1x"
                  >
                    <svg
                      aria-hidden="true"
                      class="chakra-icon chakra-select__icon css-13otjrl"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      role="presentation"
                      stroke="currentColor"
                      stroke-width="0"
                      style="color: currentColor; width: 1em; height: 1em;"
                      viewBox="0 0 24 24"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M12 14L8 10H16L12 14Z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="css-pu04yu"
            />
            <p
              class="chakra-text css-3z0v2v"
            >
              showing 1 to 1 of 1 entries
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
